src/serverless-openapi-typescript.ts

```diff
@@ -223,9 +223,9 @@ export default class ServerlessOpenapiTypeScript {
         if (customTags) openApi.tags = openApi.tags.concat(customTags)
 
         Object.values(openApi.paths).forEach(path => {
-            Object.values(path).forEach(method => {
+            Object.entries(path).forEach(([methodName, method]) => {
                 const httpEvent = this.functions[method.operationId]?.events?.find(
-                    (e: ApiGatewayEvent) => e.http
+                    (e: ApiGatewayEvent) => (e.http as any).method === methodName
```

Comment:
Typescript-wise `e.http` could be string so TS cried but I see in the next lines we used it as an object so I did that.